### PR TITLE
Hpcdb - modw etl

### PR DESCRIPTION
**Pull request by plessbd**
## Description

As part of the move to the "cloud" database structure (separation of jobfacts into job records and job tasks) this change moves part of the ingestion process from the current php (https://github.com/ubccr/xdmod/tree/xdmod7.0/classes/OpenXdmod/Ingestor/Hpcdb) to use the ETLv2 process

This currently will not work until updates to the ETL process have been merged in.  This is to have others look at the changes to see if anything was missed.

Even after merge this will not impact any current systems as these code changes make a new schema to validate against before making this the "current" process

## Motivation and Context

As part of the changes required for the collection of "cloud" statistics and the creation of job_record and job_task tables this has been implemented to consolidate code and processes.

## Tests performed

Using subset of data ran standard ingest then new hpcdb ETLv2
ETLv2 ingestion was to a new schema
Validated the results with table validator script

$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t hosts
2017-07-31 11:23:10 [notice] Compare tables src=modw.hosts, dest=modw_cloud.hosts
2017-07-31 11:23:10 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t account
2017-07-31 11:23:15 [notice] Compare tables src=modw.account, dest=modw_cloud.account
2017-07-31 11:23:15 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t fieldofscience
2017-07-31 11:23:19 [notice] Compare tables src=modw.fieldofscience, dest=modw_cloud.fieldofscience
2017-07-31 11:23:19 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t allocationonresource
2017-07-31 11:23:23 [notice] Compare tables src=modw.allocationonresource, dest=modw_cloud.allocationonresource
2017-07-31 11:23:23 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t peopleunderpi
2017-07-31 11:23:26 [notice] Compare tables src=modw.peopleunderpi, dest=modw_cloud.peopleunderpi
2017-07-31 11:23:26 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t piperson
2017-07-31 11:23:29 [notice] Compare tables src=modw.piperson, dest=modw_cloud.piperson
2017-07-31 11:23:29 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t principalinvestigator
2017-07-31 11:23:34 [notice] Compare tables src=modw.principalinvestigator, dest=modw_cloud.principalinvestigator
2017-07-31 11:23:34 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t systemaccount
2017-07-31 11:23:37 [notice] Compare tables src=modw.systemaccount, dest=modw_cloud.systemaccount
2017-07-31 11:23:37 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t queue
2017-07-31 11:23:42 [notice] Compare tables src=modw.queue, dest=modw_cloud.queue
2017-07-31 11:23:42 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t request
2017-07-31 11:23:45 [notice] Compare tables src=modw.request, dest=modw_cloud.request
2017-07-31 11:23:45 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t resourcetype
2017-07-31 11:23:49 [notice] Compare tables src=modw.resourcetype, dest=modw_cloud.resourcetype
2017-07-31 11:23:49 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t serviceprovider
2017-07-31 11:23:53 [notice] Compare tables src=modw.serviceprovider, dest=modw_cloud.serviceprovider
2017-07-31 11:23:53 [notice] Identical
$
$ ## Ignored columns are null (limitation in table checker)
$
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t allocationbreakdown -x alloc_limit -x used_allocation
2017-07-31 11:24:03 [notice] Compare tables src=modw.allocationbreakdown, dest=modw_cloud.allocationbreakdown
2017-07-31 11:24:03 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t fieldofscience_hierarchy -x division_id -x division_description -x description2
2017-07-31 11:24:07 [notice] Compare tables src=modw.fieldofscience_hierarchy, dest=modw_cloud.fieldofscience_hierarchy
2017-07-31 11:24:07 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t person -x prefix -x url -x birth_month -x birth_day -x department -x title -x citizenship_country_id -x ts -x ts_ts -x status -x middle_name -x email_address --ignore-column-type
2017-07-31 11:24:11 [notice] Compare tables src=modw.person, dest=modw_cloud.person
2017-07-31 11:24:11 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t organization -x organizationtype_id -x url -x phone -x nsf_org_code -x amie_name -x country_id -x state_id -x latitude -x longitude
2017-07-31 11:24:15 [notice] Compare tables src=modw.organization, dest=modw_cloud.organization
2017-07-31 11:24:15 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t resource_allocated -x end_date_ts
2017-07-31 11:24:19 [notice] Compare tables src=modw.resource_allocated, dest=modw_cloud.resource_allocated
2017-07-31 11:24:19 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t resourcespecs -x comments -x end_date_ts
2017-07-31 11:24:23 [notice] Compare tables src=modw.resourcespecs, dest=modw_cloud.resourcespecs
2017-07-31 11:24:23 [notice] Identical
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t allocation -x allocation_type_id -x long_name -x short_name
2017-07-31 11:24:27 [notice] Compare tables src=modw.allocation, dest=modw_cloud.allocation
2017-07-31 11:24:27 [notice] Identical
$
$ ## Column count ignored due to required updates
$
$ sudo /usr/share/xdmod/tools/dev/verify_table_data.php -s modw -d modw_cloud -t jobhosts --ignore-column-count
2017-07-31 11:24:36 [notice] Compare tables src=modw.jobhosts, dest=modw_cloud.jobhosts
2017-07-31 11:24:36 [notice] Identical
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [x] All new and existing tests passed.

### CODE REVIEW

File: configuration/etl/etl.d/hpcdb-modw-cloud.json [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    "#": "New job ingestion from mod_hpcdb to modw_cloud",
+
+    "hpcdb-jobs": [{
+        "name": "hpcdb-account-ingest",
+        "class": "DatabaseIngestor",
+        "namespace": "ETL\\Ingestor",
+        "options_class": "IngestorOptions",
+        "definition_file": "hpcdb-modw-cloud/account.json",
+        "description": "Ingest HPCDB account records from hpcdb",
+        "enabled": true,
```
Comment (jtpalmer) [inline configuration/etl/etl.d/hpcdb-modw-cloud.json]: Is it possible to make these enabled by default?

Comment (plessbd) [inline configuration/etl/etl.d/hpcdb-modw-cloud.json]: Steve is going to be doing that in a different pull request, I will update these accordingly

File: classes/ETL/Ingestor/HpcdbHostsIngestor.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Ingestor for Value Analytics people data.
```
Comment (jpwhite4) [inline classes/ETL/Ingestor/HpcdbHostsIngestor.php]: This comment looks incorrect, unless "Value Analytics people" is what the kids today are calling hostnames.

File: configuration/etl/etl_tables.d/hpcdb-modw-cloud/organization.json [hunk 3]
```diff
@@ -0,0 +1,174 @@
+{
+    "table_definition": {
+        "name": "organization",
```
Comment (jpwhite4) [inline configuration/etl/etl_tables.d/hpcdb-modw-cloud/organization.json]: Don't forget to change this to a $ref

File: configuration/etl/etl_sql.d/hpcdb-modw-cloud/job_record_hpc_post_ingest_updates.sql [hunk 4]
```diff
@@ -0,0 +1,95 @@
+-- ------------------------------------------------------------------------------------------
+-- Post-job ingestion updates
+--
+-- We assume that the wallduration and end_time_ts haven both been calculated:
+--
+-- end_time_ts = EXTRACT(epoch FROM end_time)
+--
```
Comment (plessbd) [inline configuration/etl/etl_sql.d/hpcdb-modw-cloud/job_record_hpc_post_ingest_updates.sql]: but if I dont add commneted out code how do I do version control? :)

Comment (smgallo) [inline configuration/etl/etl_sql.d/hpcdb-modw-cloud/job_record_hpc_post_ingest_updates.sql]: By blaming it on me. :) Remove the commented out code and include a reference as to where it is calculated.

**Comment by plessbd** [comment]
replaced by #349, #340, #339

**Pull request closed without merging**